### PR TITLE
CHANGES file for Chapel 1.24

### PR DESCRIPTION
**Pull request by bradcray**
This is the CHANGES.md file for Chapel 1.24.  Thanks to everyone for all the hard work (creating this file, but mostly doing the changes themselves)!

### CODE REVIEW

**Comment by bradcray** [comment]
@ronawho: I think this is ready for a round of your feedback

File: CHANGES.md [hunk 1]
```diff
@@ -1,10 +1,413 @@
...
+  - removed Python 2 dependences in favor of Python 3
+  - improved several launcher choices and options
+
+Packaging / Configuration Changes
+---------------------------------
+* a C++11 compiler is now required to build the Chapel compiler
+* added `bundled` options for several `CHPL_*` environment variables
+* added a Lua/Lmod module file for Chapel on HPE Cray XC and EX systems
+* expanded support for HPE Cray EX systems  
+  (see https://chapel-lang.org/docs/1.24/platforms/cray.html#getting-started-with-chapel-on-hpe-cray-ex-systems)
```
Comment (gbtitus) [inline CHANGES.md:53]: Sorry, I should have remembered that I'd sent you this link when I updated the docs yester-eve. The link for this is now https://chapel-lang.org/docs/1.24/platforms/cray.html#getting-started-with-chapel-on-cray-xc-or-hpe-cray-ex-systems. (Just the tag part changed, after the '#' character.)

**Comment by vasslitvinov** [comment]
We could add the `forallLoops` primer and task intents in the `taskParallel` primer.

**Comment by vasslitvinov** [comment]
Also - we may want some of these:
* add `implements` keyword under "Syntactic / Naming Changes"
* "specified non-nilable declarations for if/while in the spec" - under "Documentation"

Re "added initial support for defining constrained generic interfaces" - it is more than just "defining".  To me, dropping "defining" would do it.

In "set CHPL_OPTIMIZE/DEBUG based on user settings, not to match the runtime", should it be `CHPL_DEBUG` ? (still fits in 80 columns)

**Comment by vasslitvinov** [comment]
@bradcray:

```
* added task intents to the 'Task Parallelism' primer
  (see https://chapel-lang.org/docs/1.24/primers/taskParallel.html#task-intents)
* added a primer for 'forall' loops
  (see https://chapel-lang.org/docs/1.24/primers/forallLoops.html)
```

**Pull request merged**